src/extension.ts

```diff
@@ -20,62 +20,322 @@ import * as vscode from 'vscode';
+let outputChannelMap : Map<string, vscode.OutputChannel>;
+let curlCommand : string = '/bin/sh';
+let curlOption : string = '-c';
+
+// This extension offers basic integration with Camel-K (https://github.com/apache/camel-k) on two fronts.
 
 export function activate(context: vscode.ExtensionContext) {
 
-	outputChannel = vscode.window.createOutputChannel("Camel-K");
+	outputChannelMap = new Map();
+
+	const osType = platform();
+	if (osType === 'win32') {
+		curlCommand = 'cmd';
+		curlOption = '/c';
+	}
+
+	// process the workspace setting indicating whether we should use the proxy or CLI
+	let proxySetting = vscode.workspace.getConfiguration().get('camelk.integrations.useProxy');
+	if (proxySetting) {
+		useProxy = true;
+	}
+	camelKIntegrationsProvider.setUseProxy(useProxy);
+
+	vscode.workspace.onDidChangeConfiguration(() => {
+		let proxySetting = vscode.workspace.getConfiguration().get('camelk.integrations.useProxy');
+		if (proxySetting) {
+			useProxy = true;
+		} else {
+			useProxy = false;
+		}
+		camelKIntegrationsProvider.setUseProxy(useProxy);
+		camelKIntegrationsProvider.refresh();
+	});	
+	
+	mainOutputChannel = vscode.window.createOutputChannel("Camel-K");
+	mainOutputChannel.show();
 
 	// create the integrations view
 	vscode.window.registerTreeDataProvider('camelk.integrations', camelKIntegrationsProvider);
 
-	// create the two integration view actions -- refresh and remove
+	// create the integration view action -- refresh
 	vscode.commands.registerCommand('camelk.integrations.refresh', () => camelKIntegrationsProvider.refresh());
-	vscode.commands.registerCommand('camelk.integrations.remove', (node: TreeNode) => {
-		if (node) {
-			let commandString = 'kamel delete "' + node.label + '"';
-			console.log('Command string: ' + commandString);
-			child_process.exec(commandString, (error, stdout, stderr) => {
-				if (error) {
-					console.error(`exec error: ${error}`);
-					return;
-				}
-				console.log(`stdout: ${stdout}`);
-				console.log(`stderr: ${stderr}`);
-			});
+
+	// create the integration view action -- remove
+	vscode.commands.registerCommand('camelk.integrations.remove', async (node: TreeNode) => {
+		if (node && node.label) {
+			let integrationName : string = node.label;
+			if (useProxy) {
+				utils.shareMessage(mainOutputChannel, 'Removing ' + integrationName + ' via Kubernetes Rest Delete');
+				await stopIntegrationViaRest(integrationName).catch( (error) => {
+					utils.shareMessage(mainOutputChannel, `rest error: ${error}`);
+				});
+			} else {
+				utils.shareMessage(mainOutputChannel, 'Removing ' + integrationName + ' via Kamel executable Delete');
+				let commandString = 'kamel delete "' + integrationName + '"';
+				child_process.exec(commandString, (error, stdout, stderr) => {
+					if (error) {
+						utils.shareMessage(mainOutputChannel, `exec error: ${error}`);
+						return;
+					}
+					if (stdout) {
+						// utils.shareMessage(mainOutputChannel, `stdout: ${stdout}`);
```

Comment:
why it is commented out?
Either remove the if and the commented code or uncomment.